routes/projects/update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
+ *   files: [ <FileObj1>, <FileObj2>... ] required
```

Comment:
It should basically represent whatever is in the user's fs (on the thimble side) at the moment this request is sent. So values can range between `[ ]` to `[ file1, file2, file3... file100 ]`